api/src/main/java/shop/ink3/api/payment/entity/Payment.java

```diff
@@ -57,6 +57,10 @@ public class Payment {
     @Column(nullable = true)
     private LocalDateTime approvedAt;
 
+    public void setOrder(Order order) {
+        this.order = order;
+    }
+
     public void setDiscountPrice(Integer discountPrice) {
         this.discountPrice = discountPrice;
     }
```

Comment:
엔티티에서의 setter는 지양하는 것이 좋다고 합니다! 
해당 메서드의 네이밍이 의도를 파악하기 어렵고, 어디서든 값의 수정이 가능하도록 열려있기 때문입니다!

따라서 생성자에 빌더를 붙여 사용하거나, 메서드의 네이밍을 명확히 해주시는건 어떨까요?? setOrder() 메서드도 동일합니다!
```suggestion
    @Builder
    public Payment(Integer discountPrice) {
        this.discountPrice = discountPrice;
    }
```
또는
```suggestion 
    public void updateDiscountPrice(Integer discountPrice) {
        this.discountPrice = discountPrice;
    }
```

Reply:
set 메소드를 사용하는 것보다 메소드에 네이밍에 정확한 의도를 담는 것이 더 좋을것 같네요!
order에 대해서는 set 자체를 제거해서 detached문제에 대한 위험성을 제거했고
setDiscountPrice와 setUsedPointPrice는 assignPrice로 변경해서 메소드 네이밍을 변경했습니다!